plugins/downloader/back.js

```diff
@@ -28,6 +32,37 @@ function handle(win) {
 				console.log("Unknown action: " + action);
 		}
 	});
+	
+	// Save the songInfo when it updates
+	getSongInfo(win)(song=>{
+		songInfo = song;
+	})
+
+	ipcMain.on('add-metadata', (event, filePath) => {
+		try {
+			const songBuffer = fs.readFileSync(filePath);
+			const coverBuffer = songInfo.image.toPNG();
+			const writer = new ID3Writer(songBuffer);
+			
+			// Create the metadata tags
+			writer.setFrame('TIT2', songInfo.title)
+				.setFrame('TPE1', [songInfo.artist])
+				.setFrame('APIC', {
+					type: 3,
+					data: coverBuffer,
+					description: ''
+				});
+			writer.addTag();
+
+			// Overwrite the old file, with the new file with metadata
```

Comment:
Note: instead of writing the file, reading it again into a buffer and finally re-writing it, we can simply pass the buffer along and write the file once at the end!